### ISSUE TITLE
Error message `connect(): bad family Please make sure Pieces OS is running`

### ISSUE THREAD

**Issue by engram-design** [pieces-app/plugin_sublime#176]
With everything all installed, and many restarts of Sublime, Pieces OS and the Mac itself, I can't seem to get Sublime hooked up with Pieces.

Opening the console in Sublime shows:

```
connect(): bad family
Please make sure Pieces OS is running
```

I can see the context menu in Sublime, but everything is greyed out.

<img width="610" alt="image" src="https://github.com/user-attachments/assets/a7b542ab-7e7c-4fd7-af18-b15f0ce70e7a">

I've tried debugging this, and seems like an issue with the websocket stuff. I can confirm the URL being used is `http://localhost:1000/.well-known/health` which works in my browser. I'm not a python expert by any means, but happy to debug if you let me know where to look!

**Comment by veics** [comment]
I'm having the same issue. Today I noticed it.

**Comment by Bishoy-at-pieces** [comment]
> I'm having the same issue. Today I noticed it.
> I'm running 10.1.8 on macOS Sequoia..... uhm, this is the difference so far :)

Was it running before?

**Comment by veics** [comment]
> > I'm having the same issue. Today I noticed it.
> > I'm running 10.1.8 on macOS Sequoia..... uhm, this is the difference so far :)
> 
> Was it running before?

Yes, it was running fine. Sadly, I don't remember when I used it last time, but I think two days ago or so.
In the meantime, I upgraded my macOS to Sequoia.

**Comment by veics** [comment]
The Pieces menu is not greyed out anymore, but it doesn't seem to work.

my settings:
```
{
  "host": "http://127.0.0.1:1000",
  "llm_provider": "ollama",
  "ollama_host": "http://localhost:11434",
  "ollama_model": "starcoder2:3b"
}
```

Console output when reloading the plugin:
```
connect(): bad family
connect(): bad family
Pieces OS version: 10.1.8
Plugin version: 1.3.0
```

**Comment by Bishoy-at-pieces** [comment]
Open the settings using 
`Pieces:Open Pieces Settings`

**Comment by Bishoy-at-pieces** [comment]
I want to verify that it is working before closing the issue

**Comment by veics** [comment]
> I want to verify that it is working before closing the issue

![image](https://github.com/user-attachments/assets/70301c17-78f3-4e14-8e3c-0fa8ec74620b)

**Comment by Bishoy-at-pieces** [comment]
Thank you I will take a look

**Comment by Bishoy-at-pieces** [comment]
@engram-design can you run any command? if so then you should have an issue 
and in the message that saying I don't have a context it is fixed already in #172 and I am planning to tag a release

**Comment by engram-design** [comment]
Running version 10.1.8 of Pieces OS. Changing to a loopback address doesn't work for me:

```
{
    "host": "http://127.0.0.1:1000",
    "model":"GPT-4o Chat Model",
    "auto_start_pieces_os":true,
    "snippet.autocomplete":true,
}
```

But I can't run any command. In the command palette, this is all I have available: 
<img width="353" alt="image" src="https://github.com/user-attachments/assets/45b47ee1-122f-487b-b3ef-ee6a88251578">

Have of course tried restarting, reloading, etc.

**Comment by engram-design** [comment]
Ah, having said that, I noticed that a `LSP-intelephense` package was installed and throwing some errors, removing that and everything seems to be okay! 
<img width="594" alt="image" src="https://github.com/user-attachments/assets/be6b89db-3958-483c-96d6-63dcb0736195">

<img width="560" alt="image" src="https://github.com/user-attachments/assets/3f82cf35-d999-481a-ba53-353092f9638d">

Maybe unrelated, but I get an error when selecting "Start conversation about the current item"

```
raceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python38/sublime_plugin.py", line 1687, in run_
    return self.run(edit, **args)
TypeError: run() missing 1 required positional argument: 'pieces_query'
```

If I run Ask Copilot > Modify I can at least enter text, but get:

<img width="323" alt="image" src="https://github.com/user-attachments/assets/e5ed8ff6-6904-4230-8f11-8b5a152500a4">

I can confirm I can use "GPT-4o Chat Model" (no quota issues due to subscription) and things are working in Pieces OS.

**Comment by Bishoy-at-pieces** [comment]
> Ah, having said that, I noticed that a `LSP-intelephense` package was installed and throwing some errors, removing that and everything seems to be okay! 
> <img width="594" alt="image" src="https://github.com/user-attachments/assets/be6b89db-3958-483c-96d6-63dcb0736195">
> 
> <img width="560" alt="image" src="https://github.com/user-attachments/assets/3f82cf35-d999-481a-ba53-353092f9638d">
> 
> Maybe unrelated, but I get an error when selecting "Start conversation about the current item"
> 
> ```
> raceback (most recent call last):
>   File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python38/sublime_plugin.py", line 1687, in run_
>     return self.run(edit, **args)
> TypeError: run() missing 1 required positional argument: 'pieces_query'
> ```
> 
> If I run Ask Copilot > Modify I can at least enter text, but get:
> 
> <img width="323" alt="image" src="https://github.com/user-attachments/assets/e5ed8ff6-6904-4230-8f11-8b5a152500a4">
> 
> I can confirm I can use "GPT-4o Chat Model" (no quota issues due to subscription) and things are working in Pieces OS.
> 
> Using the command palette "Ask about the current file" once again throws a `connect(): bad family` error with no result.

Resolved in #178

**Comment by Bishoy-at-pieces** [comment]
Tagged a release all issues in that issues should be solved before testing verify that the version is 3.3.4
